controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -221,23 +224,50 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
 
 	// check request status already contains the name of the resource. if not, add it.
 	if r.StorageClassRequest.Spec.Type == "blockpool" {
+		// initialize in-memory structs
+		r.cephRadosNamespace = &rookCephv1.CephBlockPoolRadosNamespace{}
+		r.cephRadosNamespace.Namespace = r.OperatorNamespace
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
 		r.cephBlockPool.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephBlockPool" {
-				r.cephBlockPool.Name = res.Name
-				break
-			}
+
+		// check if a CephBlockPoolRadosNamespace resource exists for the desired storageconsumer and storageprofile.
+		cephRadosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err := r.Client.List(
+			r.ctx,
+			cephRadosNamespaceList,
+			client.InNamespace(r.OperatorNamespace),
+			client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)})
```

Comment:
1. We have a util func on the reconciler to handle common cases, including the injection of the common context. There is no reason to not use it here
2. Code in this controller isolates closing parentheses for function calls on a new line to make future diffs for call alteration easier. 

```suggestion
		err := r.list(
			cephRadosNamespaceList,
			client.InNamespace(r.OperatorNamespace),
			client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)},
		)
```

Reply:
Done.